compiler/grainformat/reformat.re

```diff
@@ -847,7 +856,12 @@ and print_constant = (c: Parsetree.constant) => {
     | PConstNumber(PConstNumberRational(n, d)) =>
       Printf.sprintf("%s/%s", n, d)
     | PConstBytes(b) => Printf.sprintf("%s", b)
-    | PConstChar(c) => Printf.sprintf("'%s'", String.escaped(c))
+    | PConstChar(c) =>
+      if (String.length(c) > 0) {
+        Printf.sprintf("\'%s\'", Char.escaped(c.[0]));
```

Comment:
I don't understand string encodings, but can the Char in the `c` string take up more than 1 index? (e.g. will an emoji split between index 0 and 1 in the string?)

Reply:
I don't know.   I guess we're into the world of unicode now and what the String function returns...

Reply:
Resolved by simply reproducing the original source code for constants

Reply:
But will fail on char constants until #930 is resolved

Reply:
Taken a different approach now.